Public/Src/Cache/ContentStore/Distributed/NuCache/ContentLocationDatabase.cs

```diff
@@ -665,42 +677,52 @@ public void ContentTouched(OperationContext context, ShortHash hash, UnixTime ac
         }
 
         /// <summary>
-        /// Serialize a given <paramref name="entry"/> into a byte stream.
+        /// Uses an object pool to fetch a serializer and feed it into the serialization function.
         /// </summary>
-        protected byte[] Serialize(ContentLocationEntry entry)
+        /// <remarks>
+        /// We explicitly take and pass the instance as parameters in order to avoid lambda capturing.
+        /// </remarks>
+        protected byte[] SerializeWithPool<T>(T instance, Action<T, BuildXLWriter> serializeFunc)

```

Comment:
Is this used by methods other than the ContentLocationEntry serialization method? If not, make private.

Reply:
It is used in the child class to have access to the serialization pool of the ContentLocationDatabase

---
In reply to: [301862091](https://github.com/microsoft/BuildXL/pull/571#discussion_r301862091) [](ancestors = 301862091)